safekeeper/src/timeline_manager.rs

```diff
@@ -533,6 +543,11 @@ impl Manager {
 
         if removal_horizon_segno > self.last_removed_segno {
             // we need to remove WAL
+            let Ok(timeline_gate_guard) = self.tli.gate.enter() else {
```

Comment:
The WAL removal doesn't respect the cancellation token. Should it?

Reply:
I didn't do cancellation in this one because it is only doing local filesystem metadata operations, so should be fast (and during shutdown, I'd rather let it finish freeing up local space before proceeding)